include/zephyr/drivers/counter.h

```diff
@@ -210,35 +223,106 @@ struct counter_config_info {
 	uint8_t channels;
 };
 
+/** @brief Alarm callback
+ *
+ * @param dev       Pointer to the device structure for the driver instance.
+ * @param chan_id   Channel ID.
+ * @param ticks     Counter value that triggered the alarm.
+ * @param user_data User data.
+ */
+typedef void (*counter_alarm_callback_64_t)(const struct device *dev, uint8_t chan_id,
+					    uint64_t ticks, void *user_data);
+
+/** @brief Alarm callback structure.
+ */
+struct counter_alarm_cfg_64 {
```

Comment:
would it be possible to keep a single struct backward compatible with anonymous unions? Same for `struct counter_top_cfg`

```
struct counter_alarm_cfg {
    union {
       uint32_t ticks;
#ifdef CONFIG_COUNTER_64BITS
       uint64_t ticks_64;
#endif
     }
    union {
       counter_alarm_callback_t callback;
#ifdef CONFIG_COUNTER_64BITS
       counter_alarm_callback_64_t callback_64;
#endif
     }
     void *user_data;
     uint32_t flags;
}
```